### PR TITLE
Use common fatal error handler in FMC and RT FW.

### PR DESCRIPTION
**Pull request by bluegate010**

### CODE REVIEW

**Review by rusty1968 — DISMISSED** [review]

File: drivers/src/hand_off.rs [hunk 1]
```diff
@@ -443,13 +443,6 @@ impl FirmwareHandoffTable {
         valid
     }
 }
-/// Report a non fatal firmware error and halt.
-#[allow(clippy::empty_loop)]
-pub fn report_handoff_error_and_halt(msg: &str, code: u32) -> ! {
-    crate::cprintln!("Handoff Error: {} 0x{:08X}", msg, code);
-    report_fw_error_non_fatal(code);
```
[the remark this answers is not in the document]
Comment (bluegate010) [reply drivers/src/hand_off.rs:450]: Have noted as such in the readme.

File: fmc/src/hand_off.rs [hunk 2]
```diff
@@ -44,47 +44,42 @@ impl HandOff {
...
+                cprintln!("Handoff : FMC CDI: {:?}", key_id as u8);
                 key_id
             }
-            _ => caliptra_common::report_handoff_error_and_halt(
-                "Invalid KeySlot DV Entry",
-                caliptra_error::CaliptraError::FMC_HANDOFF_INVALID_PARAM.into(),
-            ),
+            _ => {
+                cprintln!("Invalid KeySlot KV Entry");
+                handle_fatal_error(CaliptraError::FMC_HANDOFF_INVALID_PARAM.into())
```
[the remark this answers is not in the document]
Comment (bluegate010) [reply fmc/src/hand_off.rs:63]: I wasn't able to find in the FMC README where handoff error handling was described. Have added a note there that all boot failures are fatal.

File: fmc/src/hand_off.rs [hunk 3]
```diff
@@ -44,70 +44,59 @@ impl HandOff {
...
-                    "Invalid CDI DV handle",
-                    caliptra_error::CaliptraError::FMC_HANDOFF_INVALID_PARAM.into(),
-                )
-            });
+        let ds: DataStore =
+            Self::fht(env)
+                .fmc_cdi_kv_hdl
+                .try_into()
+                .unwrap_or_else(|e: CaliptraError| {
+                    cprintln!("Invalid CDI KV handle");
```
Comment (jhand2) [inline fmc/src/hand_off.rs:52]: nit (and similarly for other areas in this file)
```suggestion
                    cprintln!("[fmc] Invalid CDI KV handle");
```

File: runtime/src/main.rs [hunk 4]
```diff
@@ -42,20 +42,18 @@ pub extern "C" fn entry_point() -> ! {
             // treat global exception as a fatal error
             match e {
                 CaliptraError::RUNTIME_GLOBAL_EXCEPTION => handle_fatal_error(e.into()),
-                _ => caliptra_common::report_handoff_error_and_halt(
-                    "Runtime can't load drivers",
```
Comment (jhand2) [inline runtime/src/main.rs:46]: similarly I think `[rt]` prefix will make these messages easier to read.

**Review by jhand2 — DISMISSED** [review]

File: fmc/src/hand_off.rs (hunk 3 above)
Comment (bluegate010) [reply fmc/src/hand_off.rs]: Done.

File: runtime/src/main.rs (hunk 4 above)
Comment (bluegate010) [reply runtime/src/main.rs:46]: Done

**Review by jhand2 — DISMISSED** [review]

**Review by jhand2 — APPROVED** [review]

**Review by jhand2 — APPROVED** [review]

**Review by rusty1968 — APPROVED** [review]

**Pull request merged**